### PR TITLE
Regen

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @EPashkin @sdroege

### CODE REVIEW

File: src/cairo_interaction.rs [hunk 1]
```diff
@@ -68,9 +68,8 @@ impl ContextExt for Context {
         }
     }
 
-    fn cairo_surface_create_from_pixbuf<'a, P: Into<Option<&'a Window>>>(pixbuf: &Pixbuf, scale: i32, for_window: P) -> Option<Surface> {
+    fn cairo_surface_create_from_pixbuf(pixbuf: &Pixbuf, scale: i32, for_window: Option<&Window>) -> Option<Surface> {
```
Comment (EPashkin) [inline src/cairo_interaction.rs]: @sdroege Not related to this PR, you don't know why `gdk::Window` don't used with `IsA` ?

**Comment by EPashkin** [comment]
👍

File: src/cairo_interaction.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/cairo_interaction.rs]: That's a good question actually.

Comment (EPashkin) [reply src/cairo_interaction.rs]: Oh, this not generated code, so it related to this PR actually 😉

**Comment by EPashkin** [comment]
@GuillaumeGomez not better add PR to examples first?

**Comment by GuillaumeGomez** [comment]
@EPashkin I'll do it next.

File: src/cairo_interaction.rs (hunk 1 above)
Comment (sdroege) [reply src/cairo_interaction.rs]: Should be `IsA`

File: src/auto/device.rs [hunk 2]
```diff
@@ -206,8 +206,7 @@ impl Device {
     }
 
     #[cfg_attr(feature = "v3_20", deprecated)]
-    pub fn grab<'a, P: IsA<Window>, Q: Into<Option<&'a Cursor>>>(&self, window: &P, grab_ownership: GrabOwnership, owner_events: bool, event_mask: EventMask, cursor: Q, time_: u32) -> GrabStatus {
-        let cursor = cursor.into();
+    pub fn grab<P: IsA<Window>>(&self, window: &P, grab_ownership: GrabOwnership, owner_events: bool, event_mask: EventMask, cursor: Option<&Cursor>, time_: u32) -> GrabStatus {
```
Comment (sdroege) [inline src/auto/device.rs]: Should be `IsA<Cursor>` or not?

File: src/event.rs [hunk 3]
```diff
@@ -248,9 +248,9 @@ impl Event {
         }
     }
 
-    pub fn set_screen<'a, T: Into<Option<&'a Screen>>>(&mut self, screen: T) {
+    pub fn set_screen(&mut self, screen: Option<&Screen>) {
```
Comment (sdroege) [inline src/event.rs]: These should also all be `IsA<_>` or not?

**Comment by sdroege** [comment]
Looks good apart from the missing `IsA<_>`

File: src/event.rs (hunk 3 above)
Comment (EPashkin) [reply src/event.rs]: Seems it final
https://github.com/gtk-rs/gdk/blob/47bdccab08a8366788b74e2ab1bc97f6f7269d81/src/auto/app_launch_context.rs#L46

Comment (sdroege) [reply src/event.rs]: That's good

Comment (sdroege) [reply src/event.rs]: Should double-check for all these types then :)

File: src/auto/device.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/auto/device.rs]: That's automatic generation so if problem there is, it lies in gir.

**Comment by EPashkin** [comment]
Only maybe non-final type used in manual code is `FrameClock`,
not sure that it need `FrameClockExtManual` or `final_type = true`.

**Comment by GuillaumeGomez** [comment]
I'll merge this one then meanwhile. Waiting for the next releases. :)

**Pull request merged**